index.d.ts

```diff
@@ -0,0 +1,373 @@
+   *
+   * @default undefined
+   */
+  sourceMap?: boolean | string;
+
+  /**
+   * Includes the contents in the source map information.
+   *
+   * @default false
+   */
+  sourceMapContents?: boolean;
+
+  /**
+   * Embeds the source map as a data URI.
+   *
+   * @default false
+   */
+  sourceMapEmbed?: boolean;
+
+  /**
+   * The value will be emitted as `sourceRoot` in the source map information.
+   *
+   * @default undefined
+   */
+  sourceMapRoot?: string;
+
+  /**
+   * Callbacks for handling any `@warn` or `@debug` messages
+   *
+   * @default undefined
+   */
+  logger?: Logger;
+}
+
+export interface SassException extends Error {
+  /**
+   * The error message.
+   */
+  message: string;
+
+  /**
+   * The formatted error.
+   */
+  formatted: string;
+
+  /**
+   * The line number of error.
+   */
+  line: number;
+
+  /**
+   * The column number of error.
+   */
+  column: number;
+
+  /**
+   * The status code.
+   */
+  status: number;
+
+  /**
+   * The filename of error.
+   *
+   * In case file option was not set (in favour of `data`), this will reflect the value `stdin`.
+   */
+  file: string;
+}
+
+export interface SourceLocation {
+  /**
+   * The 0-indexed offset of the location
+   */
+  offset: number;
+
+  /**
+   * The line number of the location
+   */
+  line: number;
+
+  /**
+   * The 0-indexed column of the location
+   */
+  column: number;
+}
+
+export interface SourceSpan {
+  /**
+   * The text between `start` and `end` locations.
+   * This is from the exact line and column
+   */
+  text: string;
+
+  /**
+   * The start location of the `@warn` or `@debug` message
+   */
+  start: SourceLocation;
+
+  /**
+   * The end location of the `@warn` or `@debug` message
+   */
+  end: SourceLocation;
+
+  /**
+   * The location of the file with the message
+   */
+  url: string;
+
+  /**
+   * The text from `start` to `end`. However, unlike `text`, it outputs each full line
+   */
+  context: string;
+}
+
+export interface Result {
+  /**
+   * The compiled CSS.
+   *
+   * Write this to a file, or serve it out as needed.
+   */
+  css: Buffer;
+
+  /**
+   * The source map.
+   */
+  map?: Buffer;
+  stats: {
+    /**
+     * The path to the scss file, or `data` if the source was not a file.
+     */
+    entry: string;
+
+    /**
+     * `Date.now()` before the compilation.
+     */
+    start: number;
+
+    /**
+     * `Date.now()` after the compilation.
+     */
+    end: number;
+
+    /**
+     * `end - start`
+     */
+    duration: number;
+
+    /**
+     * Absolute paths to all related files in no particular order.
+     */
+    includedFiles: string[];
+  };
+}
+
+export function render(
+  options: Options,
+  callback: (exception: SassException, result: Result) => void
+): void;
+export function renderSync(options: Options): Result;
+
+export namespace types {
+  abstract class SassType {}
+
+  interface Null extends SassType {
+    NULL: Null;
+  }
+
+  const Null: Null;
```

Comment:
Because you're specifying the `Null` interface separately from the corresponding constant, I'm not sure if TypeScript will understand that `types.Null.NULL` is an instance of `types.Null`. You may need to just make this a `class` without a (visible) constructor.